munit/shared/src/main/scala/munit/MUnitRunner.scala

```diff
@@ -246,9 +246,15 @@ class MUnitRunner(val cls: Class[_ <: Suite], newInstance: () => Suite)
       case NonFatal(ex) =>
         trimStackTrace(ex)
         val cause = ex match {
-          case e: ExecutionException
-              if "Boxed Exception" == e.getMessage() &&
-                e.getCause() != null =>
+          case e: ExecutionException if e.getCause() != null && {
+                e.getMessage() match {
+                  // NOTE(olafur): these exceptions appear when we await on
+                  // futures. We unwrap these exception in order to provide more
+                  // helpful error messages.
+                  case "Boxed Exception" | "Boxed Error" => true
+                  case _                                 => false
```

Comment:
do we need the check on `getMessage`? I think something like

https://github.com/scala/scala/blob/ba9701059216c629410f4f23a2175d20ad62484b/src/compiler/scala/tools/nsc/util/Exceptional.scala#L19-L27

would work here (and it's nice to handle also other cases)

Reply:
Nice! That ends up being way more general, the old solution was not nice at all